spec/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/refresher_spec.rb

```diff
@@ -14,25 +16,42 @@
   it "tests the refresh" do
     2.times do
       # Storing ems as a variable and passing it directly reduces the complexity of the runtime. Which appears to increase runtime performance.
-      mgmt = ems
-      mgmt.refresh
-      mgmt.reload
-
-      assert_ems_counts(mgmt)
-      assert_specific_vm(mgmt)
-      assert_specific_resource_group(mgmt, '29b1dd25de2d40b5ae5bd5f719f30db8', 'camc-test')
-      assert_specific_security_group(mgmt, 'r014-e4be0c69-6df6-4464-a9bc-384e4179ea1b', 'backup-deglazed-bagful-deflation', 'r014-0fa2acc6-2a41-4f2b-9c89-bcea07cdcbc3')
-      assert_specific_cloud_volume_type(mgmt, 'general-purpose', 'tiered')
-      assert_specific_cloud_subnet(mgmt, '0757-ef523a2f-5356-42ff-8a78-9325509465b9', 'r014-0fa2acc6-2a41-4f2b-9c89-bcea07cdcbc3', 'us-east-1')
-      assert_specific_floating_ip(mgmt, 'r014-1892eee4-79ab-4d26-8efd-4c5759fc03fc', '150.239.208.80', '0777_a9ee9e6a-231a-4a3c-b9cb-fc83d25114a2')
-      assert_vm_labels(mgmt, '0777_f73e8687-3813-465f-99df-ba6e4ee8f289', 4)
+      with_vcr { ems.refresh }
+      ems.reload
+
+      assert_ems_counts(ems)
+      assert_specific_vm(ems)
+      assert_specific_resource_group(ems, '29b1dd25de2d40b5ae5bd5f719f30db8', 'camc-test')
+      assert_specific_security_group(ems, 'r014-e4be0c69-6df6-4464-a9bc-384e4179ea1b', 'backup-deglazed-bagful-deflation', 'r014-0fa2acc6-2a41-4f2b-9c89-bcea07cdcbc3')
+      assert_specific_cloud_volume_type(ems, 'general-purpose', 'tiered')
+      assert_specific_cloud_subnet(ems, '0757-ef523a2f-5356-42ff-8a78-9325509465b9', 'r014-0fa2acc6-2a41-4f2b-9c89-bcea07cdcbc3', 'us-east-1')
+      assert_specific_floating_ip(ems, 'r014-1892eee4-79ab-4d26-8efd-4c5759fc03fc', '150.239.208.80', '0777_a9ee9e6a-231a-4a3c-b9cb-fc83d25114a2')
+      assert_vm_labels(ems, '0777_f73e8687-3813-465f-99df-ba6e4ee8f289', 4)
+    end
+  end
+
+  context "targeted refresh" do
+    before { with_vcr { ems.refresh } }
+
+    context "vm target" do
+      let(:target) { ems.vms.find_by(:ems_ref => "0757_81687d4a-4676-4eeb-9fd7-55f9c7fffb69") }
+
+      it "doesn't impact other inventory" do
+        before_targeted_refresh = serialize_inventory
+
+        with_vcr("vm_target") { EmsRefresh.refresh(target) }
+
+        after_targeted_refresh = serialize_inventory
+
+        assert_inventory_not_changed(before_targeted_refresh, after_targeted_refresh)
```

Comment:
TODO this would be cool called as a block that serialized first, yielded, then serialized after so this was just `assert_inventory_not_changed { with_vcr("vm_target") { EmsRefresh.refresh(target) } }`

Reply:
https://github.com/ManageIQ/manageiq/pull/21281